docs/extensions/index.md

```diff
@@ -52,3 +52,15 @@ Extensions which undergo a [formal review](/extensions/lifecycle.md#formal-revie
 
 >> See: [Automated Distribution](/extensions/publish.md#automated-distribution)
 
+## Interacting with core
+
+There are a number of ways in which extensions can interact with core. These are supported to greater or lesser degrees.
+
+Fully supported methods are:
```

Comment:
The combination of statements feels a little confusing: "These are supported to greater or lesser degrees." followed by "Fully supported methods are:".

Proposals:
* __Easy__: Just kill the second sentence.
* __Difficult__:  Kill the second sentence. Additionally, update each bullet-point with a comment or rating about support/maturity.

Reply:
I think the below are fully supported - I feel like there are lot of things in the report interface that are semi supported

Reply:
Actually WRT hooks - they are supported to the extent they will continue to exist - but the code around them might change

Reply:
FWIW, the sentences seem fairly straightforward to me. Maybe there would be some mileage in using recommended, in place of / in addition to supported.

"There are a number of ways in which an extension *could* interact with core. However, not all of them are recommended. For example, you should try avoid the use of BAOs and SQL statements whenever possible. The following are the recommend ways of interacting with core. Sticking to these methods increases the likelihood your extension will survive upgrades and be compatible with other extensions."